### PR TITLE
Add bbes for binary operators

### PR DESCRIPTION
**Pull request by SasinduDilshara**
Add bbes for binary operators and unary operators

Fixes https://github.com/ballerina-platform/ballerina-dev-website/issues/9324

### CODE REVIEW

**Comment by SasinduDilshara** [comment]
> @SasinduDilshara can you please send two separate PRs for the two BBEs?

updated

File: examples/binary-operators/binary_operators.bal [hunk 1]
```diff
@@ -0,0 +1,67 @@
+import ballerina/io;
+
+public function main() {
+    int a = 10;
+    int b = 20;
+
+    // Addition
```
Comment (MaryamZi) [inline examples/binary-operators/binary_operators.bal:7]: These comments aren't good enough. We can mention at least what is expected as the operands.

File: examples/binary-operators/binary_operators.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# Binary operators
+
+Ballerina provides several binary operators to perform operations on various data types, such as integers, floats, booleans, and strings. Binary arithmetic operators (+, -, *, /, %) allow for basic mathematical calculations between two values. Comparison operators (==, !=, <, >, <=, >=) are used to compare two values and produce a boolean result.
```
Comment (MaryamZi) [inline examples/binary-operators/binary_operators.md:3]: - on values of various types
- remove comma before such as
- use backticks with operators
- don't introduce terms like "comparison operators" which don't come from the spec
- why are equality checks mentioned here?

File: examples/binary-operators/binary_operators.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+
+In addition, Ballerina supports binary logical operators like && and || for combining boolean expressions, and bitwise operators (&, |, ^, <<, >>, >>>) for manipulating individual bits in integers. These operators enable efficient and flexible handling of binary data and logical operations in a concise manner.
```
Comment (MaryamZi) [inline examples/binary-operators/binary_operators.md:5]: - In addition, Ballerina supports -> Ballerina also supports
- What does "combining boolean expressions" mean? We have to be careful with how we describe things. We have to be accurate. Either extract from spec descriptions (preferred) or use standard descriptions.
- Last sentence isn't necessary here either.

File: examples/binary-operators/binary_operators.md [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+
+::: code binary_operatos.bal :::
+
+::: out binary_operatos.out :::
```
Comment (MaryamZi) [inline examples/binary-operators/binary_operators.md:9]: Typos!

File: examples/binary-operators/binary_operators.metatags [hunk 5]
```diff
@@ -0,0 +1,2 @@
+description: This BBE introduces the binary operators associated with ballerina types.
```
Comment (MaryamZi) [inline examples/binary-operators/binary_operators.metatags:1]:
```suggestion
description: This BBE introduces binary operators in Ballerina.
```

File: examples/binary-operators/binary_operators.metatags [hunk 6]
```diff
@@ -0,0 +1,2 @@
...
+keywords: ballerina, ballerina by example, bbe, operators, binary operators, bitwise operators, int, float, string, boolean, +, -, /, %, *, , <, >, <=, >=, >>, <<, >>>, ||, &&, |, &, ^
```
Comment (MaryamZi) [inline examples/binary-operators/binary_operators.metatags:2]: IIRC, we used to have a limit on the number here. @sm1990 @keizer619 any concerns with these?

File: examples/binary-operators/binary_operators.md (hunk 2 above)
Comment (SasinduDilshara) [reply examples/binary-operators/binary_operators.md:3]: Used the words logical operators, bitwise operations.
Removed the `==` and `!=`

File: examples/binary-operators/binary_operators.md (hunk 3 above)
Comment (SasinduDilshara) [reply examples/binary-operators/binary_operators.md:5]: Used the words logical operators, bitwise operations.

File: examples/binary-operators/binary_operators.metatags (hunk 6 above)
Comment (sm1990) [reply examples/binary-operators/binary_operators.metatags:2]: That is not an issue anymore.

File: examples/binary-operators/binary_operators.bal [hunk 7]
```diff
@@ -0,0 +1,65 @@
+import ballerina/io;
+
+public function main() {
+    int a = 10;
+    int b = 20;
+
+    // Adds two integers.
```
Comment (MaryamZi) [inline examples/binary-operators/binary_operators.bal:7]:
```suggestion
    // Add two integers.
```

Change everywhere.

Also better to assign to a variable and then print to demonstrate static type.

File: examples/binary-operators/binary_operators.bal [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+    io:println(a + b);
+    // Subtracts two integers.
+    io:println(a - b);
+    // Multiplies two integers.
+    io:println(a * b);
+    // Divides two integers.
+    io:println(b / a);
+    // Computes the modulus of two integers.
```
Comment (MaryamZi) [inline examples/binary-operators/binary_operators.bal:15]: integer remainder is what the spec uses.

File: examples/binary-operators/binary_operators.bal [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+    io:println(b % a);
+    // Compares two integers.
```
Comment (MaryamZi) [inline examples/binary-operators/binary_operators.bal:17]: Compare how?

File: examples/binary-operators/binary_operators.md [hunk 10]
```diff
@@ -0,0 +1,10 @@
+# Binary operators
+
+Ballerina provides several binary operators to perform operations on values of various types such as integers, floats, booleans, and strings. Binary arithmetic operators (`+`, `-`, `*`, `/`, `%`) allow for basic mathematical calculations between two values. The operators for comparing values (`<`, `>`, `<=`, `>=`) produce a boolean result.
```
Comment (MaryamZi) [inline examples/binary-operators/binary_operators.md:3]:
```suggestion
Ballerina provides several binary operators to perform operations on values of various types such as integers, floats, booleans, and strings. Binary arithmetic operators (`+`, `-`, `*`, `/`, `%`) allow basic mathematical calculations between two values. The operators for comparing values (`<`, `>`, `<=`, `>=`) produce a boolean result.
```

Comment (MaryamZi) [reply examples/binary-operators/binary_operators.md:3]: "comparing" needs to be updated.

File: examples/binary-operators/binary_operators.md [hunk 11]
```diff
@@ -0,0 +1,10 @@
...
+
+Ballerina also supports binary logical operators (`&&`, `||`) for performing logical operations on boolean values, as well as bitwise operators (`&`, `|`, `^`, `<<`, `>>`, `>>>`) for manipulating individual bits in integer values.
```
Comment (MaryamZi) [inline examples/binary-operators/binary_operators.md:5]:
```suggestion
Ballerina also supports binary logical operators (`&&` and `||`) to perform logical operations on boolean values and bitwise operators (`&`, `|`, `^`, `<<`, `>>`, and `>>>`) to manipulate bits in integer values.
```

Change the previous para also.

File: examples/binary-operators/binary_operators.md (hunk 10 above)
Comment (SasinduDilshara) [reply examples/binary-operators/binary_operators.md]: Updated with `The operators to compare values`

File: examples/binary-operators/binary_operators.bal [hunk 12]
```diff
@@ -0,0 +1,106 @@
...
+    int remainder = b % a;
+    io:println(remainder);
+
+    // Check if `a` is less than `b`.
+    boolean isLessThan = a < b;
+    io:println(isLessThan);
+
+    // Check if `a` is greater than or equal to `b`.
+    boolean isGreaterOrEqual = a >= b;
+    io:println(isGreaterOrEqual);
```
Comment (MaryamZi) [inline examples/binary-operators/binary_operators.bal:33]: Why no examples for non-numeric types?

File: examples/binary-operators/binary_operators.bal [hunk 13]
```diff
@@ -0,0 +1,115 @@
...
+
+public function main() {
+    int a = 10;
+    int b = 20;
+
+    // Add two integers.
+    int sum = a + b;
+    io:println(sum);
+
+    // Subtract two integers.
```
Comment (MaryamZi) [inline examples/binary-operators/binary_operators.bal:11]:
```suggestion
    // Perform integer subtraction.
``` 

So have to change others accordingly.

File: examples/binary-operators/binary_operators.bal [hunk 14]
```diff
@@ -0,0 +1,115 @@
...
+
+    // Divide two integers.
+    int quotient = b / a;
+    io:println(quotient);
+
+    // Compute the integer remainder.
+    int remainder = b % a;
+    io:println(remainder);
+
+    // `>`, `<`, `>=` and `<=` are use to perform the relative order of two values.
```
Comment (MaryamZi) [inline examples/binary-operators/binary_operators.bal:27]:
```suggestion
    // `>`, `<`, `>=`, and `<=` are use to test the relative order of two values.
```

File: examples/binary-operators/binary_operators.bal [hunk 15]
```diff
@@ -0,0 +1,115 @@
...
+    // Check if `a` is less than `b`.
+    boolean isLessThan = a < b;
+    io:println(isLessThan);
+
+    // Check if `a` is greater than or equal to `b`.
+    boolean isGreaterOrEqual = a >= b;
```
Comment (MaryamZi) [inline examples/binary-operators/binary_operators.bal:33]:
```suggestion
    boolean isGreaterThanOrEqual = a >= b;
```

File: examples/binary-operators/binary_operators.bal [hunk 16]
```diff
@@ -0,0 +1,115 @@
...
+    io:println(isLessOrEqual);
+
+    // Check if `e` is greater than `f`.
+    boolean isGreaterThan = e > f;
+    io:println(isGreaterThan);
+
+    boolean c = true;
+    boolean d = false;
+
+    // Perform logical AND between two boolean values.
```
Comment (MaryamZi) [inline examples/binary-operators/binary_operators.bal:66]:
```suggestion
    // Perform the logical AND operation between two boolean values.
```

File: examples/binary-operators/binary_operators.bal [hunk 17]
```diff
@@ -0,0 +1,115 @@
...
+    boolean logicalAnd = c && d;
+    io:println(logicalAnd);
+
+    // Perform logical OR between two boolean values.
+    boolean logicalOr = c || d;
+    io:println(logicalOr);
+
+    // `false` is considered less than `true`.
```
Comment (MaryamZi) [inline examples/binary-operators/binary_operators.bal:74]:
```suggestion
    // For the `<` operator, `d < c` evaluates to `true` when `d` is `false` and `c` is `true`.
```

File: examples/binary-operators/binary_operators.md [hunk 18]
```diff
@@ -0,0 +1,9 @@
+# Binary operators
+
+Ballerina provides several binary operators to perform operations on values of various types such as integers, floats, booleans, and strings. Binary arithmetic operators (`+`, `-`, `*`, `/`, `%`) allow basic mathematical calculations between two values. The operators to compare values (`<`, `>`, `<=`, `>=`) produce a boolean result.
```
Comment (MaryamZi) [inline examples/binary-operators/binary_operators.md:3]: Fix operator lists to use ", and"

Comment (MaryamZi) [reply examples/binary-operators/binary_operators.md:3]: Doesn't cover + for string.

Comment (SasinduDilshara) [reply examples/binary-operators/binary_operators.md:3]: Since it is describe in the example, Ino no need to add in here

**Review by gimantha — DISMISSED** [review]

**Comment by MaryamZi** [comment]
@SasinduDilshara can you please resolve the conflict?

**Comment by SasinduDilshara** [comment]
> @SasinduDilshara can you please resolve the conflict?

updated

**Review by gimantha — APPROVED** [review]

**Pull request merged**